### PR TITLE
feat: add tensorrt_yolox package

### PR DESCRIPTION
**Pull request by manato**
## Description

This PR aims to add the `tensorrt_yolox` package into the series of perception modules in autoware.universe.
The `tensorrt_yolox` package that works with autoware has been published in [separated repository](https://github.com/tier4/tensorrt_yolox). This PR integrates the package's contents into autoware.universe.

## Related links

- [TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-17617)
- https://github.com/tier4/tensorrt_yolox

## Tests performed

```
# build
colcon build --symlink-install --cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=Release --packages-up-to tensorrt_yolox

# run
ros2 run tensorrt_yolox tensorrt_yolox_node_exe --ros-args -p model_path:=install/tensorrt_yolox/share/tensorrt_yolox/data/yolox-tiny.onnx -p label_path:=install/tensorrt_yolox/share/tensorrt_yolox/data/label.txt -r /tensorrt_yolox/in/image:=/image_raw
```

## Notes for reviewers

I've added small modifications  listed below from [the original repository](https://github.com/tier4/tensorrt_yolox)
- treating labels as case insensitive
- treating the `PERSON`  label in the same way as the `PEDESTRIAN` label for COCO dataset compatibility 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/tensorrt_yolox/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+project(tensorrt_yolox)
+
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+endif()
```
Comment (yukkysaito) [inline perception/tensorrt_yolox/CMakeLists.txt:11]: It is better to use autoware_cmake
https://github.com/autowarefoundation/autoware.universe/blob/main/perception/multi_object_tracker/CMakeLists.txt#L4-L5

**Comment by wep21** [comment]
I think it's better to separate `tensorrt_yolox`, `tensorrt_common`, `cuda_utils` because another package can use common functions in `tensorrt_common`, `cuda_utils`.

**Comment by manato** [comment]
> I think it's better to separate `tensorrt_yolox`, `tensorrt_common`, `cuda_utils` because another package can use common functions in `tensorrt_common`, `cuda_utils`.

@wep21 
I really appreciate your suggestion. I agree with you, so let me confirm where to put `tensorrt_common` and `cuda_utils` for further modification. Do you think `autoware.universe/common/` is the proper place to put them? If you think a different directory is preferable, I would appreciate it if you let me know.

File: perception/tensorrt_yolox/CMakeLists.txt (hunk 1 above)
Comment (manato) [reply perception/tensorrt_yolox/CMakeLists.txt]: @yukkysaito 
Thank you very much for the comment. I replaced the lines you mentioned with the macro provided by `autoware_cmake` in [this commit](a314e8d791111645d68057deefcfbb8147d5dab7).
I would appreciate it if you check that my modification meets your intention.

Comment (yukkysaito) [reply perception/tensorrt_yolox/CMakeLists.txt]: @manato LGTM

**Review by yukkysaito — APPROVED** [review]
LGTM

**Comment by wep21** [comment]
formatをautowareにあわせます。

**Comment by miursh** [comment]
@manato You should add your name to maintainer in package.xml. 
Also, could you fix CI error?

File: perception/tensorrt_yolox/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+target_link_libraries(${PROJECT_NAME}
+  ${tensorrt_common_LIBRARIES}
+)
+
+target_compile_definitions(${PROJECT_NAME} PRIVATE
+  TENSORRT_VERSION_MAJOR=${TENSORRT_VERSION_MAJOR}
+)
+
+ament_auto_add_library(yolox_single_image_inferece_node SHARED
```
Comment (yukke42) [inline perception/tensorrt_yolox/CMakeLists.txt:73]:
```suggestion
ament_auto_add_library(yolox_single_image_inference_node SHARED
```
There are some typos of `inference`.

Comment (manato) [reply perception/tensorrt_yolox/CMakeLists.txt]: @yukke42 
I really appreciate your support. I correct these typos in https://github.com/autowarefoundation/autoware.universe/pull/2370/commits/b96564189a298dcd06b8ceef50e502e6f2838b75

**Comment by manato** [comment]
> @manato You should add your name to maintainer in package.xml. Also, could you fix CI error?

@miursh 
Thank you very much for pointing this out. I completely missed it. I added maintainer descriptions in https://github.com/autowarefoundation/autoware.universe/pull/2370/commits/4fa36aedce9c110d5b03859e7e5633619bb59499

Also, the CI errors seem to be gone with the kind help of @yukke42 and @wep21 . I really appreciate their grateful support.

**Pull request merged**